modules/core/src/test/java/org/apache/ignite/internal/processors/cache/consistency/AbstractFullSetReadRepairTest.java

```diff
@@ -270,20 +270,20 @@
      * @param data Data.
      */
     protected void check(ReadRepairData data, IgniteIrreparableConsistencyViolationException e, boolean evtRecorded) {
-        Collection<?> irreparableKeys = e != null ? e.irreparableKeys() : null;
+        Collection<Object> irreparableKeys = e != null ? e.irreparableKeys() : null;
 
         if (e != null) {
-            Collection<?> repairableKeys = e.repairableKeys();
+            Collection<Object> repairableKeys = e.repairableKeys();
 
             if (repairableKeys != null)
                 assertTrue(Collections.disjoint(repairableKeys, irreparableKeys));
 
-            Collection<?> expectedToBeIrreparableKeys = data.data.entrySet().stream()
+            Collection<Object> expectedToBeIrreparableKeys = data.data.entrySet().stream()
                 .filter(entry -> !entry.getValue().repairable)
                 .map(Map.Entry::getKey)
                 .collect(Collectors.toSet());
 
-            assertEquals(expectedToBeIrreparableKeys, irreparableKeys);
+            assertEqualsCollectionsIgnoringOrder(expectedToBeIrreparableKeys, irreparableKeys);
```

Comment:
Because equals fails on different collection types.
Possible solution is to call Collections.unmodifiableMap(..) over expectedToBeIrreparableKeys, but this will look odd ...

Reply:
Just don't wrap irreparable keys to unmodifiable collection?